src/Microsoft.TemplateEngine.TemplateLocalizer.Core/TemplateString.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.TemplateEngine.TemplateLocalizer.Core
+{
+    /// <summary>
+    /// Represents a string in template.json file that needs to be localized.
+    /// </summary>
+    internal struct TemplateString : IEquatable<TemplateString>
```

Comment:
Please follow this guideline: https://docs.microsoft.com/en-us/dotnet/api/system.iequatable-1?view=net-5.0#remarks